.github/workflows/api_reference.yaml

```diff
@@ -0,0 +1,41 @@
+name: api_reference
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+    # Only run api_reference action when a header file has been modified.
+    paths:
+      - '**.hpp'
+
+jobs:
+  doxygen:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          path: src/Fast-DDS-QoS-Profiles-Manager
+
+      - name: Install doxygen and cmake
+        run: |
+          sudo apt install doxygen cmake
```

Comment:
There is a common action to install apt dependencies in eProsima/eProsima-CI#4. I think we should use that action even though we have to wait till that PR is merged.